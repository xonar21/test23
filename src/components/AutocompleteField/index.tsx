import React, { useState, useEffect } from "react";
import { FormControl, TextField, CircularProgress, Autocomplete, styled } from "@mui/material";
import { useDispatch } from "react-redux";
import { SelectListAction, SelectListSelector, SelectListsActions } from "~/store";

interface AutocompleteFieldProps {
  field: any;
  fieldValue: any;
  fieldKey: string;
  setValue: (newValue: any) => void;
  value: any;
  dinamicTitle: string;
  t: any;
  openModal: boolean;
  selectList: any;
  paramsRequest: any;
}

const AutocompleteField: React.FC<AutocompleteFieldProps> = ({
  field,
  fieldValue,
  fieldKey,
  setValue,
  value,
  dinamicTitle,
  t,
  openModal,
  selectList,
  paramsRequest,
}) => {
  const dispatch = useDispatch();
  const [page, setPage] = useState(1);
  const [loading, setLoading] = useState(false);
  const [totalPages, setTotalPages] = useState(0);
  const [searchValue, setSearchValue] = useState("");
  const actionKey: SelectListAction = field.action;
  const action = SelectListsActions[actionKey];
  const selectkey: SelectListSelector = field.keySelect;
  const [items, setItems] = useState(selectList[selectkey].items);
  const StyledListbox = styled("ul")({
    "&::-webkit-scrollbar": {
      width: "4px",
    },
    "&::-webkit-scrollbar-thumb": {
      backgroundColor: "rgba(0, 48, 92, 1)",
      borderRadius: "4px",
    },
    "&::-webkit-scrollbar-track": {
      backgroundColor: "rgba(230, 230, 230, 1)",
    },
  });

  useEffect(() => {
    const fetchData = async () => {
      if (field.action && !field.parent) {
        await dispatch(
          action(
            paramsRequest(page, 20, {
              fieldName: "value",
              values: [searchValue],
            }),
          ),
        );
      }
    };

    fetchData();
  }, [page, searchValue]);

  useEffect(() => {
    setTotalPages(selectList[selectkey].totalPages);

    if (selectList[selectkey].totalPages === 1) {
      setItems(selectList[selectkey].items);

      return;
    }

    setItems((prevItems: any) => [...prevItems, ...selectList[selectkey].items]);
  }, [selectList[selectkey].items]);

  useEffect(() => {
    const fetchData = async () => {
      if (field.parent && value[field.parent]) {
        const paramsRequest = (
          id: string,
          number?: number,
          size?: number,
          filters?: object,
          sortField?: string,
          sortOrder?: string,
        ) => {
          return {
            id,
            number,
            size,
            filters,
            sortField,
            sortOrder,
          };
        };

        await dispatch(
          action(
            paramsRequest(value[field.parent], page, 20, {
              fieldName: "value",
              values: [searchValue],
            }),
          ),
        );
      }
    };

    fetchData();
  }, [value[field.parent]]);

  useEffect(() => {
    if (field.parent) {
      setItems([]);
    }
  }, [openModal]);

  return (
    <FormControl fullWidth>
      <Autocomplete
        id="demo-autocomplete"
        options={items}
        value={fieldValue}
        getOptionLabel={option => {
          if (dinamicTitle === t("edit")) {
            let itemValue = "";
            items.forEach((e: any) => {
              if (e.key === option) {
                itemValue = e.value;
              }
            });
            return itemValue;
          }
          return option ? option.value : "";
        }}
        onChange={(e, newValue) => {
          setValue({ ...value, [fieldKey]: newValue?.key });
        }}
        onInputChange={(event, value, reason) => {
          if (reason === "reset") {
            setPage(1);
          }
          if (reason === "input") {
            setSearchValue(value);
            setPage(1);
            setItems([]);
          }
          if (reason === "clear") {
            setPage(1);
            setSearchValue("");
          }
          if (value === "") {
            setPage(1);
            setSearchValue("");
          }
        }}
        ListboxComponent={StyledListbox}
        ListboxProps={{
          onScroll: event => {
            const target = event.target as HTMLUListElement;
            const { scrollTop, scrollHeight, clientHeight } = target;
            if (scrollHeight - scrollTop === clientHeight) {
              setPage(prevPage => {
                if (prevPage !== totalPages) {
                  return prevPage + 1;
                }
                return prevPage;
              });
            }
          },
        }}
        renderInput={params => (
          <TextField
            {...params}
            label={t(`${field.fieldName}`)}
            InputLabelProps={{ ...params.InputLabelProps }}
            InputProps={{
              ...params.InputProps,
              endAdornment: (
                <>
                  {loading ? <CircularProgress size={20} /> : null}
                  {params.InputProps.endAdornment}
                </>
              ),
            }}
          />
        )}
        renderOption={(props, option) => <li {...props}>{option.value}</li>}
      />
    </FormControl>
  );
};

export default AutocompleteField;
