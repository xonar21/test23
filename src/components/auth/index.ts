export * from "./auth-button";
export * from "./auth-form";
export * from "./auth-provider";
