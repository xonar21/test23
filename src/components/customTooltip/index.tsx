import React, { useMemo } from "react";
import { ColDef, ITooltipParams } from "ag-grid-community";
import { Box } from "@mui/material";

type CustomTooltipParams = ITooltipParams & {
  color: string;
  colDef: ColDef;
  shouldShowTooltip: boolean;
  rowData: any;
};
export default (props: CustomTooltipParams) => {
  const data = useMemo(() => props.api.getDisplayedRowAtIndex(props.rowIndex!)!.data, []);

  if (props && props.column) {
    if ("colId" in props.column) {
      const cellSelector = `.ag-row[row-id="${props.rowIndex}"] .ag-cell[col-id="${props.column.getColId()}"]`;
      const cellElement = document.querySelector(cellSelector);
      if (cellElement && cellElement) {
        if (cellElement.scrollWidth === cellElement.clientWidth) {
          return null;
        }
      }
    }
  }

  const subProperties = (data: any, props: any) => {
    const [firstField, secondField] = props.colDef.field.split(".");

    if (secondField) {
      return data[firstField][secondField];
    }

    return data[firstField];
  };
  return (
    <>
      {props.colDef.field ? (
        <Box
          sx={{
            backgroundColor: props.color || "white",
            border: "1px solid rgba(0, 48, 92, 1)",
            minWidth: "100px",
            borderRadius: "3px",
            padding: "5px",
            transition: "all 1s linear",
            maxWidth: "300px",
            whiteSpace: "pre-wrap !important",
            overflowWrap: "break-word !important",
            boxSizing: "border-box !important",
          }}
        >
          {subProperties(data, props)}
          {/* {data[props.colDef.field]} */}
        </Box>
      ) : (
        <></>
      )}
    </>
  );
};
