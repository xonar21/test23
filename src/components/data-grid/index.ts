export { default as DataGrid } from "./DataGrid";
export * from "./configs";
