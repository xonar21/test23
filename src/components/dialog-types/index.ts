export { default as DialogTypes } from "./dialog-types";
export * from "./dialog-types";
