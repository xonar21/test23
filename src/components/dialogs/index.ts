export * from "./prompt-dialog";
export * from "./post-form-dialog";
export * from "./dialog-provider";
