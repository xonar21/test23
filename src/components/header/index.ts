export { default as Header } from "./Header";
export { default as HeaderPublic } from "./HeaderPublic";
