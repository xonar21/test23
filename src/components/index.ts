export * from "./auth";
export * from "./breadcrumbs";
export * from "./data-grid";
export * from "./demo";
export * from "./dialogs";
export * from "./dialog-types";
export * from "./formik";
export * from "./header";
export * from "./mui";
export * from "./navigation-drawer";
export * from "./page-loading-indicator";
export * from "./seo";
export * from "./snackbar-provider";
