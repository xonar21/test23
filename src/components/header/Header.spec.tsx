import { render, fireEvent, testIds, Locale } from "~/shared";
import { AppHeader } from "~/components";

describe("Header component", () => {
  it("should render without crashing", () => {
    const { asFragment } = render(<AppHeader />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should toggle drawer", () => {
    const { getByTestId } = render(<AppHeader />);
    const button = getByTestId(testIds.components.header.buttons.drawerToggler);
    const drawer = getByTestId(testIds.components.navigationDrawer.root);
    fireEvent.click(button);
    const expected = "true";
    const actual = drawer.attributes.getNamedItem("aria-expanded")?.value;
    expect(actual).toEqual(expected);
  });

  it("should handle redirect on navigation link click", () => {
    const { getAllByTestId } = render(<AppHeader />);
    const link = getAllByTestId(testIds.components.navigationDrawer.navigationLink)[0];
    fireEvent.click(link);
  });

  it("should change locale without crashing", () => {
    const { getByTestId, getAllByTestId } = render(<AppHeader />);

    const localeMenu = getByTestId(testIds.components.header.buttons.localeMenu);

    for (let i = 0; i < Object.keys(Locale).length; i++) {
      fireEvent.click(localeMenu);
      const localeItem = getAllByTestId(testIds.components.header.localeItem)[i];
      fireEvent.click(localeItem);
    }
  });
});
