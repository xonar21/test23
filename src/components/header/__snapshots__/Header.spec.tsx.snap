// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render without crashing 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
    data-testid="header"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <button
        aria-label="menu"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
        data-testid="header-drawer-toggle-button"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
      >
        brandName
      </div>
      <button
        aria-label="dark-mode-toggle"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
        data-testid="header-theme-toggle-button"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="Brightness3Icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9 2c-1.05 0-2.05.16-3 .46 4.06 1.27 7 5.06 7 9.54 0 4.48-2.94 8.27-7 9.54.95.3 1.95.46 3 .46 5.52 0 10-4.48 10-10S14.52 2 9 2z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-haspopup="true"
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-bkseul-MuiButtonBase-root-MuiButton-root"
        data-testid="header-locale-menu"
        tabindex="0"
        type="button"
      >
        language
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1y942vo-MuiButtonBase-root-MuiButton-root"
        data-testid="header-auth-button"
        tabindex="0"
        type="button"
      >
        auth.login
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
  <div
    aria-expanded="false"
    class="MuiDrawer-root MuiDrawer-docked css-1frs1v9-MuiDrawer-docked"
    data-testid="navigation-drawer"
    links="[object Object],[object Object]"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
    >
      <ul
        class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
        >
          <div
            aria-label="routes.home"
            class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
            data-mui-internal-clone-element="true"
            data-testid="navigation-link"
            role="button"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="HomeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                routes.home
              </span>
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
        >
          <div
            aria-label="routes.posts"
            class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
            data-mui-internal-clone-element="true"
            data-testid="navigation-link"
            role="button"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArticleIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-5 14H7v-2h7v2zm3-4H7v-2h10v2zm0-4H7V7h10v2z"
              />
            </svg>
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                routes.posts
              </span>
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
